File: andyrb/mood.rb
# frozen_string_literal: true

require 'paint'
# Convenience methods to add a colored star based on the "mood" method called.
module Mood
  def self.happy(message = nil)
    message = yield if block_given? && message.nil?
    Paint['*', :green] + ' ' + message
  end

  def self.neutral(message = nil)
    message = yield if block_given? && message.nil?
    Paint['*', :yellow] + ' ' + message
  end

  def self.sad(message = nil)
    message = yield if block_given? && message.nil?
    Paint['*', :red] + ' ' + message
  end

  def self.colorful(color, message = nil)
    message = yield if block_given? && !message
    Paint['*', color.to_sym] + ' ' + message
  end

  class << self
    def [](color, message = nil)
      Paint['*', color.to_sym] + ' ' + message
    end
  end
end

File: andypy/mood.py
import platform
from termcolor import colored


class Mood:

    """Class to replace the old Color class that uses static methods instead of string conditionals.
    Also, because it is using static methods, it no longer requires class instantiation."""

    @staticmethod
    def happy():
        """Prints a green star on unix, a generic star on windows."""
        system = platform.system

        if system is not "Windows":
            return colored("*", "green")
        else:
            return "*"

    @staticmethod
    def neutral():
        """Prints a yellow star on unix, a generic star on windows."""
        system = platform.system

        if system is not "Windows":
            return colored("*", "yellow")
        else:
            return "*"

    @staticmethod
    def sad():
        """Prints a red star on unix, a generic star on windows."""
        system = platform.system

        if system is not "Windows":
            return colored("*", "red")
        else:
            return "*"
